src/goto-instrument/contracts/contracts.cpp

```diff
@@ -734,6 +733,29 @@ code_contractst::create_ensures_instruction(
   return std::make_pair(std::move(ensures_program), std::move(history));
 }
 
+static const code_with_contract_typet &
+get_contract(const irep_idt &function, const namespacet &ns)
+{
+  const std::string function_str = id2string(function);
+  const symbolt &function_symbol = ns.lookup(function);
+
+  const symbolt *contract_sym;
+  if(ns.lookup("contract::" + function_str, contract_sym))
+  {
+    // no contract provided in the source or just an empty assigns clause
+    return to_code_with_contract_type(function_symbol.type);
+  }
+
+  const auto &type = to_code_with_contract_type(contract_sym->type);
+  if(type != function_symbol.type)
+  {
+    throw invalid_input_exceptiont(
+      "Contract of '" + function_str + "' has different signature.");
+  }
```

Comment:
nitpick: could we cover this with a regression test?

Reply:
Done!